### PR TITLE
[script.commands] 1.1.0

### PR DESCRIPTION
**Pull request by RomanSmolka**
### Description
This is a Matrix port of the [Commands addon](https://kodi.wiki/view/Add-on:Commands) originally made by @AddonScriptorDE. The author has been inactive on both GitHub and Kodi forum since 2015.

- No functional changes have been made. 
- Language files were recreated to use the .po format
- default.py script was updated for Python 3.

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: script.commands/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Commands – Kodi Addon
+=========================
+Map multiple XBMC commands to one remote button. All availabe commands: https://kodi.wiki/view/List_of_built-in_functions
```
Comment (enen92) [inline script.commands/README.md:3]:
```suggestion
Map multiple Kodi commands to one remote button. All available commands: https://kodi.wiki/view/List_of_built-in_functions
```

File: script.commands/README.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+This is a fork of the apparently abandoned addon made by AddonScriptorDE. It has been updated to work with Kodi 19 Matrix.
```
Comment (enen92) [inline script.commands/README.md:5]:
```suggestion
This is a fork of the abandoned commands XBMC addon made by AddonScriptorDE. It has been updated to work with Kodi 19 Matrix.
```

File: script.commands/addon.xml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+       provider-name="AddonScriptorDE, Roman Smolka">
+  <requires>
+    <import addon="xbmc.python" version="3.0.0"/>
+  </requires>
+  <extension point="xbmc.python.pluginsource"
+            library="default.py">
+    <provides>executable</provides>
+  </extension>
+  <extension point="xbmc.addon.metadata">
+      <summary lang="en">Map multiple XBMC commands to one remote button. All availabe commands: https://kodi.wiki/view/List_of_built-in_functions</summary>
```
Comment (enen92) [inline script.commands/addon.xml:14]:
```suggestion
      <summary lang="en_GB">Map multiple XBMC commands to one remote button. All availabe commands: https://kodi.wiki/view/List_of_built-in_functions</summary>
```

File: script.commands/addon.xml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+      <summary lang="de">Mehrere XBMC Befehle einer Taste zuweisen. Alle verfügbaren Befehle: https://kodi.wiki/view/List_of_built-in_functions</summary>
```
Comment (enen92) [inline script.commands/addon.xml:15]:
```suggestion
      <summary lang="de_DE">Mehrere XBMC Befehle einer Taste zuweisen. Alle verfügbaren Befehle: https://kodi.wiki/view/List_of_built-in_functions</summary>
```

File: script.commands/addon.xml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+      <language></language>
```
Comment (enen92) [inline script.commands/addon.xml:16]: Remove, this only applies to media content

File: script.commands/addon.xml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+      <license>GNU GENERAL PUBLIC LICENSE. Version 2, June 1991</license>
+      <source>https://github.com/AddonScriptorDE/script.commands</source>
```
Comment (enen92) [inline script.commands/addon.xml:18]: This needs to be the new upstream location of the code

File: script.commands/addon.xml [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+      <forum>http://forum.xbmc.org/showthread.php?tid=134129</forum>
+      <website></website>
```
Comment (enen92) [inline script.commands/addon.xml:20]: drop if empty

File: script.commands/addon.xml [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+  </requires>
+  <extension point="xbmc.python.pluginsource"
+            library="default.py">
+    <provides>executable</provides>
+  </extension>
+  <extension point="xbmc.addon.metadata">
+      <summary lang="en">Map multiple XBMC commands to one remote button. All availabe commands: https://kodi.wiki/view/List_of_built-in_functions</summary>
+      <summary lang="de">Mehrere XBMC Befehle einer Taste zuweisen. Alle verfügbaren Befehle: https://kodi.wiki/view/List_of_built-in_functions</summary>
+      <language></language>
+      <license>GNU GENERAL PUBLIC LICENSE. Version 2, June 1991</license>
```
Comment (enen92) [inline script.commands/addon.xml:17]:
```suggestion
      <license>GPL-2.0-only</license>
```

File: script.commands/changelog.txt [hunk 9]
```diff
@@ -0,0 +1,9 @@
+1.0.0
```
Comment (enen92) [inline script.commands/changelog.txt:1]: Changelog is missing your new version. Also, Kodi now has the `<news>` element in addon xml to display the news regarding the latest addon version. Consider to use it.

File: script.commands/commands.list [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+Show System Info###ActivateWindow(10007)
+Show Settings###ActivateWindow(10004)
+Show Shutdown Menu###ActivateWindow(10111)
+Show Favourites###ActivateWindow(10134)
+Visualisation: Info###Action(ShowPreset)
+Visualisation: Menu###ActivateWindow(VisualisationPresetList)
+Visualisation: Lock###Action(LockPreset)
+Visualisation: Next###Action(nextpreset)
+Visualisation: Prev###Action(previouspreset)
+SimplePlaylists: Add###RunPlugin(plugin://script.simpleplaylists/?mode=addCurrentUrl)
```
Comment (enen92) [inline script.commands/commands.list:57]: I'm not aware of this addon in our repo, please drop the action

File: script.commands/commands.list [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+SimplePlaylists: View###RunPlugin(plugin://script.simpleplaylists/?mode=showPlaylists)
```
Comment (enen92) [inline script.commands/commands.list:58]: same

File: script.commands/commands.list [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+Toggle Subtitles###Action(showsubtitles)
+Cycle Languages###Action(audionextlanguage)
+Cycle Aspect Ratios###Action(aspectratio)
+Show Subtitle Delay###Action(subtitledelay)
+Show Audio Delay###Action(audiodelay)
+Toggle Audio Output###Action(audiotoggledigital)
+Take Screenshot###TakeScreenshot
+Refresh RSS feeds###RefreshRSS
+Eject Tray###EjectTray
+Play DVD###PlayDVD
```
Comment (enen92) [inline script.commands/commands.list:15]: Deprecated, use PlayDisc

File: script.commands/commands.list [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+Search Video Library###VideoLibrary.Search
+Video Partymode###PlayerControl(Partymode(video))
+Music Partymode###PlayerControl(Partymode(music))
+Rotate Picture###Action(rotate)
+Reload Skin###ReloadSkin
+Reboot System###Reboot
+ShutDown System###ShutDown
+Hibernate System###Hibernate
+Suspend System###Suspend
+Restart XBMC###RestartApp
```
Comment (enen92) [inline script.commands/commands.list:38]: Kodi

File: script.commands/commands.list [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+Quit XBMC###Quit
```
Comment (enen92) [inline script.commands/commands.list:39]: Kodi

File: script.commands/default.py [hunk 15]
```diff
@@ -0,0 +1,323 @@
+import xbmc, xbmcgui, xbmcaddon, locale, sys, re, os
+
+addonID = "script.commands"
+addon_work_folder=xbmc.translatePath("special://profile/addon_data/"+addonID)
```
Comment (enen92) [inline script.commands/default.py:4]:
```suggestion
addon_work_folder=xbmcvfs.translatePath("special://profile/addon_data/"+addonID)
```

File: script.commands/default.py [hunk 16]
```diff
@@ -0,0 +1,323 @@
...
+commandsFile=xbmc.translatePath('special://profile/addon_data/'+addonID+'/commands')
```
Comment (enen92) [inline script.commands/default.py:5]:
```suggestion
commandsFile=xbmcvfs.translatePath('special://profile/addon_data/'+addonID+'/commands')
```

File: script.commands/default.py [hunk 17]
```diff
@@ -0,0 +1,323 @@
+import xbmc, xbmcgui, xbmcaddon, locale, sys, re, os
```
Comment (enen92) [inline script.commands/default.py:1]:
```suggestion
import xbmc, xbmcgui, xbmcaddon, xbmcvfs, locale, sys, re, os
```

File: script.commands/default.py [hunk 18]
```diff
@@ -0,0 +1,323 @@
...
+
+addonID = "script.commands"
+addon_work_folder=xbmc.translatePath("special://profile/addon_data/"+addonID)
+commandsFile=xbmc.translatePath('special://profile/addon_data/'+addonID+'/commands')
+commandsList=xbmc.translatePath('special://home/addons/'+addonID+'/commands.list')
```
Comment (enen92) [inline script.commands/default.py:6]:
```suggestion
commandsList=xbmcvfs.translatePath('special://home/addons/'+addonID+'/commands.list')
```

File: script.commands/default.py [hunk 19]
```diff
@@ -0,0 +1,323 @@
...
+favsFile=xbmc.translatePath("special://profile/favourites.xml")
```
Comment (enen92) [inline script.commands/default.py:7]:
```suggestion
favsFile=xbmcvfs.translatePath("special://profile/favourites.xml")
```

File: script.commands/default.py [hunk 20]
```diff
@@ -0,0 +1,323 @@
...
+        nr=dialog.select(translation(30002), myCommandsTemp)
+        if nr>=0:
+          entry=favs[nr]
+          command = entry[entry.find("###")+3:]
+          currentCombo+=command+"#"
+          addCommandCombo(title,key)
+        else:
+          currentCombo=currentCombo[:len(currentCombo)-1]
+          if title+"###"+currentCombo not in myCommands:
+            fh = open(commandsFile+key, 'a')
```
Comment (enen92) [inline script.commands/default.py:97]:
```python
with open(commandsFile+key, 'a') as fh:
    fh.write(title+"###"+currentCombo+"\n")
```

File: script.commands/default.py [hunk 21]
```diff
@@ -0,0 +1,323 @@
...
+            fh.write(title+"###"+currentCombo+"\n")
+            fh.close()
+          commandsMain(key)
+
+def addCommandList(key):
+        if os.path.exists(commandsList):
+          favs=[]
+          fh = open(commandsList, 'r')
```
Comment (enen92) [inline script.commands/default.py:105]: same, use python's `with`

File: script.commands/default.py [hunk 22]
```diff
@@ -0,0 +1,323 @@
...
+          for temp in favs:
+            myCommandsTemp.append(temp[:temp.find("###")])
+          dialog = xbmcgui.Dialog()
+          nr=dialog.select(translation(30002), myCommandsTemp)
+          if nr>=0:
+            entry=favs[nr]
+            title = entry[:entry.find("###")]
+            command = entry[entry.find("###")+3:]
+            if title+"###"+command not in myCommands:
+              fh = open(commandsFile+key, 'a')
```
Comment (enen92) [inline script.commands/default.py:119]: same

File: script.commands/default.py [hunk 23]
```diff
@@ -0,0 +1,323 @@
...
+              fh.close()
+            commandsMain(key)
+          else:
+            addCommand(key)
+
+def addCommandFavs(key):
+        if os.path.exists(favsFile):
+          favs=[]
+          fh = open(favsFile, 'r')
+          for line in fh:
```
Comment (enen92) [inline script.commands/default.py:130]: same

File: script.commands/default.py [hunk 24]
```diff
@@ -0,0 +1,323 @@
...
+        kb = xbmc.Keyboard("", translation(30010))
+        kb.doModal()
+        if kb.isConfirmed():
+          title=kb.getText()
+          kb = xbmc.Keyboard("", translation(30009))
+          kb.doModal()
+          if kb.isConfirmed():
+            command=kb.getText()
+            if title+"###"+command not in myCommands:
+              fh = open(commandsFile+key, 'a')
```
Comment (enen92) [inline script.commands/default.py:165]: same

File: script.commands/default.py [hunk 25]
```diff
@@ -0,0 +1,323 @@
...
+                command=kb.getText()
+                newContent=""
+                fh = open(commandsFile+key, 'r')
+                for line in fh:
+                  if line.find(oldTitle+"###")==0:
+                    newContent+=line.replace(oldTitle,title).replace(oldCommand,command)
+                  else:
+                    newContent+=line
+                fh.close()
+                fh=open(commandsFile+key, 'w')
```
Comment (enen92) [inline script.commands/default.py:219]: same

File: script.commands/default.py [hunk 26]
```diff
@@ -0,0 +1,323 @@
...
+            kb = xbmc.Keyboard(title, translation(30010))
+            kb.doModal()
+            if kb.isConfirmed():
+              title=kb.getText()
+              kb = xbmc.Keyboard(command, translation(30009))
+              kb.doModal()
+              if kb.isConfirmed():
+                command=kb.getText()
+                newContent=""
+                fh = open(commandsFile+key, 'r')
```
Comment (enen92) [inline script.commands/default.py:212]: same

File: script.commands/default.py [hunk 27]
```diff
@@ -0,0 +1,323 @@
...
+            myCommandsTemp.append(temp[:temp.find("###")])
+        dialog = xbmcgui.Dialog()
+        nr=dialog.select(translation(30007), myCommandsTemp)
+        if nr>=0:
+          entry=myCommands[nr]
+          if entry.find("###")>=0:
+            title = entry[:entry.find("###")]
+            command = entry[entry.find("###")+3:]
+            newContent=""
+            fh = open(commandsFile+key, 'r')
```
Comment (enen92) [inline script.commands/default.py:240]: same

File: script.commands/default.py [hunk 28]
```diff
@@ -0,0 +1,323 @@
...
+            for line in fh:
+              if line.find(title+"###")==0:
+                pass
+              else:
+                newContent+=line
+            fh.close()
```
Comment (enen92) [inline script.commands/default.py:246]: same

File: script.commands/default.py [hunk 29]
```diff
@@ -0,0 +1,323 @@
...
+          if entry.find("###")>=0:
+            dialog = xbmcgui.Dialog()
+            pos = dialog.numeric(0, translation(30011))
+            try:
+              pos=int(pos)
+            except:
+              pos=-1
+            if int(pos)<len(myCommandsTemp) and int(pos)>0:
+              newContent=""
+              fh = open(commandsFile+key, 'r')
```
Comment (enen92) [inline script.commands/default.py:273]: same

File: script.commands/default.py [hunk 30]
```diff
@@ -0,0 +1,323 @@
...
+              for line in fh:
+                if line.find(entry)==0:
+                  pass
+                else:
+                  newContent+=line
+              fh.close()
```
Comment (enen92) [inline script.commands/default.py:279]: same

File: script.commands/default.py [hunk 31]
```diff
@@ -0,0 +1,323 @@
...
+              lineNr=1
+              for line in fh:
+                if int(pos)==lineNr:
+                  newContent+=entry+"\n"
+                  newContent+=line
+                else:
+                  newContent+=line
+                lineNr+=1
+              fh.close()
+              fh=open(commandsFile+key, 'w')
```
Comment (enen92) [inline script.commands/default.py:294]: same

File: script.commands/addon.xml [hunk 32]
```diff
@@ -0,0 +1,22 @@
...
+      <source>https://github.com/AddonScriptorDE/script.commands</source>
+      <forum>http://forum.xbmc.org/showthread.php?tid=134129</forum>
+      <website></website>
+  </extension>
```
Comment (enen92) [inline script.commands/addon.xml:21]: icon has to be specified in addon.xml (`assets`)

**Review by enen92 — CHANGES_REQUESTED** [review]
See comments

**Comment by RomanSmolka** [comment]
Thank you for the fast response, much appreciated! PR has been updated with all your suggested changes.

**Review by enen92 — APPROVED** [review]
You still have a lot of places where you open and close the filehandlers. Please consider updating the code in the future to use `with` whenever possible (that will ensure the handler is always closed).

**Comment by enen92** [comment]
Please check travis failures, you have a few files marked as executable:
https://travis-ci.com/github/xbmc/repo-scripts/builds/218919200

**Comment by RomanSmolka** [comment]
Fixed, looks ok now.

**Comment by enen92** [comment]
Yeah, sure. Not everything is bad with a complete "new beginning" we enforced in matrix, at least there's the chance to cleanup and improve on the old base :)

**Pull request merged**